### PR TITLE
AdoptAWidget: Dismissible

### LINKED ISSUE #69486: Adopt a Widget - Dismissible
The [Dismissible](https://master-api.flutter.dev/flutter/widgets/Dismissible-class.html) widget needs:

- A code / DartPad sample

For details about how to participate, see [flutter.dev/adoptawidget](https://flutter.dev/adoptawidget)

### PR DESCRIPTION
**Pull request by abd99**
Added a DartPad sample to demonstrate the use of the `Dismissible` widget.

This pull request is:

- [x] A code sample

closes #69486

### CODE REVIEW

File: packages/flutter/lib/src/widgets/dismissible.dart [hunk 1]
```diff
@@ -62,6 +62,39 @@ enum DismissDirection {
...
+///        ),
+///        key: ValueKey(items[index]),
+///        onDismissed: (direction) {
+///          setState(() {
+///            items.remove(index);
+///          });
+///        },
+///      ),
+///    );
+///  }
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/dismissible.dart:94]: The entire block has one extra space.

File: packages/flutter/lib/src/widgets/dismissible.dart [hunk 2]
```diff
@@ -62,6 +62,39 @@ enum DismissDirection {
...
+///        background: Container(
+///          color: Colors.green,
+///        ),
+///        key: ValueKey(items[index]),
+///        onDismissed: (direction) {
+///          setState(() {
+///            items.remove(index);
+///          });
+///        },
+///      ),
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/dismissible.dart:92]: We tend to not use `=>` for anything that isn't a single line, so:
```dart
///      itemBuilder: (context, index) {
///        return Dismissible(
///          ...
///        );
///      }
```

File: packages/flutter/lib/src/widgets/dismissible.dart [hunk 3]
```diff
@@ -62,6 +62,39 @@ enum DismissDirection {
...
+/// remove list items using swipe gestures.
+///
+/// ```dart
+///  List<int> items = List<int>.generate(100, (index) => index);
+///
+///  Widget build(BuildContext context) {
+///    return ListView.builder(
+///      itemCount: items.length,
+///      padding: const EdgeInsets.symmetric(vertical: 16),
+///      itemBuilder: (context, index) => Dismissible(
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/dismissible.dart:77]: Define the types in the arguments:
```suggestion
///      itemBuilder: (BuildContext context, int index) => Dismissible(
```

File: packages/flutter/lib/src/widgets/dismissible.dart [hunk 4]
```diff
@@ -62,6 +62,39 @@ enum DismissDirection {
...
+///        child: ListTile(
+///          title: Text(
+///            'Item ${items[index]}',
+///          ),
+///        ),
+///        background: Container(
+///          color: Colors.green,
+///        ),
+///        key: ValueKey(items[index]),
+///        onDismissed: (direction) {
```
Comment (shihaohong) [inline packages/flutter/lib/src/widgets/dismissible.dart:87]:
```suggestion
///        onDismissed: (DismissDirection direction) {
```

**Review by shihaohong — APPROVED** [review]
LGTM

**Comment by johnpryan** [comment]
@abd99 Thanks for participating in AdoptAWidget. We would like to send you a special thanks. If you are interested, please send an email to nyener at google dot com and provide a link to this issue.

**Pull request merged**